### DISCUSSION TITLE
How can I have subfolders in the generated folder?

### DISCUSSION THREAD

**Discussion by lhanks** [cyrilletuzi/vscode-angular-schematics#933] [Q&A] [answered]
I can't seem to find a way to get the extension to generate subfolders.  I would like to have a schematic that looks like the following:

component/file.component.ts
component/file.component.html
file.module.ts
file-routing.module.ts

Is it possible to generate this file layout?

**Comment by cyrilletuzi** [comment, marked as answer]
Hello @lhanks,

All files in the `files` folder of your custom schematic will be generated by the extension, included the ones in subfolders.

Here are the steps to match your example:
1. click on the "Create a custom schematic" button and fill the basic inputs (schematic id, name...)
2. choose "From scratch", as your scenario is an advanced one
3. in the opened `schematic.json`, change `mainFile` to `{{fileName}}.module.ts.hbs`
4. add `angular` in `features` (not essential for the example but better)
5. in the `files` folder, rename the file `{{fileName}}.module.ts.hbs` (it must match `mainFile` from step 3)
6. add all the others files and folders, to have something like this:
  - `schematics/`
    - `module-with-component/` (it will be the schematic id chosen on step 1)
      - `files/`
        - `{{fileName}}.module.ts.hbs`
        - `{{fileName}}-routing.module.ts.hbs`
        - `component/`
          - `{{fileName}}.component.ts.hbs`
          - `{{fileName}}.component.html.hbs`

That is it. You can now generate the schematic.

Be sure to have the just released version 6.15.1 of the extension installed and applied, as I ran into a few bugs when reproducing this scenario, which I fixed (it may require 2 VS Code restarts, one the the extension to update, the second one for the update to be applied).
